qcodes/instrument/base.py

```diff
@@ -38,13 +39,36 @@ class InstrumentBase(Metadatable, DelegateAttributes):
             such as channel lists or logical groupings of parameters.
             Usually populated via ``add_submodule``
     """
-    def __init__(self, name, **kwargs):
+
+    def __init__(self, name, testing=False, **kwargs):
         self.name = str(name)
+        self.testing = testing
```

Comment:
This should probably be a private attribute `_testing`

Reply:
yeah good point

Reply:
We could also make it a property that cannot be set and get rid of the is_testing method below. Whichever way you prefer but there should only be one public attribute/method